public/app/plugins/datasource/graphite/specs/store.test.ts

```diff
@@ -472,4 +477,62 @@ describe('Graphite actions', async () => {
       expect(ctx.state.target.target).toEqual(expected);
     });
   });
+
+  describe('when auto-completing over a large set of tags and metrics', () => {
+    const manyMetrics: Array<{ text: string }> = [],
+      manyTags: Array<{ text: string }> = [],
+      manyTagValues: Array<{ text: string }> = [],
+      max = 20000;
+
+    beforeEach(() => {
+      for (let i = 0; i < max; i++) {
+        manyMetrics.push({ text: `metric${i}` });
+        manyTags.push({ text: `tag${i}` });
+        manyTagValues.push({ text: `tagValue${i}` });
+      }
+      ctx.state.datasource.metricFindQuery = jest.fn().mockReturnValue(Promise.resolve(manyMetrics));
+      ctx.state.datasource.getTagsAutoComplete = jest.fn().mockReturnValue(Promise.resolve(manyTags));
+      ctx.state.datasource.getTagValuesAutoComplete = jest.fn().mockReturnValue(Promise.resolve(manyTagValues));
+    });
+
+    it('uses limited metrics and tags list', async () => {
```

Comment:
can we use `MAX_SUGGESTIONS` in this test instead of testing for `10000` so that we are not forced to change these tests if we want to change the limit?

Reply:
I didn't want to expose it just for testing. What's more I think it makes sense that the test should change when implementation changes. When MAX_SUGGESTIONS is changed I'd like to see what tests failed to be double check what scenarios are affected.

Reply:
i agree that exposing it just for testing it's not ideal, that's why I was suggesting to move the limiting logic to the UI. For instance, the autocomplete component could accept a `limit` prop and we could pass that limit to the underlying function instead of relying on a that constant. that would also enable us to write tests form the user POV (for example by testing that no matter how many items the API returns, on the screen they will appear on `limit` number of options).

but I'm also ok with this, just wanted to consider a different approach. 👍

Reply:
i agree that exposing it just for testing it's not ideal, that's why I was suggesting moving the limiting logic to the UI. For instance, the autocomplete component could accept a `limit` prop and we could pass that limit to the underlying function instead of relying on a that constant. that would also enable us to write tests from the user POV (for example by testing that no matter how many items the API returns, on the screen they will appear only `limit` number of options).

but I'm also ok with this, just wanted to consider a different approach. 👍